### PR TITLE
spatialite_tools: enable on darwin, fix cross-compilation

### PR DESCRIPTION
**Pull request by sikmir**
###### Description of changes
 * freexl: init [Packaging status](https://repology.org/project/freexl/versions)
 * librttopo: fix cross-compilation
 * libspatialite: fix cross-compilation, enable freexl
 * readosm: enable on darwin, cleanup, enable tests
 * spatialite_tools: enable on darwin, enable freexl, add version test

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross)
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg build freexl librttopo libspatialite readosm spatialite_tools

(same as automatic but without passthru tests which aren't all applicable to e.g. darwin aarch64)

File: pkgs/development/libraries/libspatialite/default.nix [hunk 1]
```diff
@@ -49,13 +49,15 @@ stdenv.mkDerivation rec {
...
-  # - check_sql_stmt_extension
   # - check_virtualtable6
   # - check_drop_rename
-  # - check_get_normal_zoom_extension_load
   doCheck = false;
 
+  preCheck = ''
+    export LD_LIBRARY_PATH=$(pwd)/src/.libs
+    export DYLD_LIBRARY_PATH=$(pwd)/src/.libs
+  '';
```
Comment (risicle) [inline pkgs/development/libraries/libspatialite/default.nix:59]: Does this signal an intention to enable the tests?

Comment (sikmir) [reply pkgs/development/libraries/libspatialite/default.nix:59]: Yes, in the future, currently there are 2 failed tests:
```
libspatialite> FAIL: check_virtualtable6
libspatialite> =========================
libspatialite> Unexpected num of rows for test 28: 0.
libspatialite> FAIL: check_drop_rename
libspatialite> =======================
libspatialite> RenameTable "table_1" error: RenameTable exception - error in trigger ISO_metadata_reference_row_id_value_insert: no such column: rowid.
```
I don't think that's something nix specific.

Comment (risicle) [reply pkgs/development/libraries/libspatialite/default.nix:59]: Wonder if it's expecting a specific sqlite version.

**Comment by risicle** [comment]
Well, `nixpkgs-review` reveals no failures on macos 10.15 for me now, but still need linux testing (which will no doubt involve qgis rebuilds :partying_face: )

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64. `pkgsMusl` & `pkgsCross.aarch64-multiplatform` variants of `spatialite_tools` build happily.

**Pull request merged**